clang/lib/AST/Interp/ByteCodeExprGen.cpp

```diff
@@ -858,8 +858,8 @@ bool ByteCodeExprGen<Emitter>::VisitStringLiteral(const StringLiteral *E) {
 
   // If the initializer string is too long, a diagnostic has already been
   // emitted. Read only the array length from the string literal.
-  unsigned N =
-      std::min(unsigned(CAT->getSize().getZExtValue()), E->getLength());
+  unsigned ArraySize = CAT->getSize().getZExtValue();
```

Comment:
So `getZExtValue()` returns `uint64_t` yet we are using `unsigned`. I know this is already an existing problem that we make varying assumptions in different places but I am wondering what the long-term plan is for this this. Forgive me if I have asked this question in another context and I just forgot.

Reply:
It's a bit tricky since `StringLiteral::getLength()` returns `unsigned` again, so we can't use the extra bytes anyway and need to cast to `unsigned` for the `std::min()`.